cmd/poseidon/main.go

```diff
@@ -155,24 +159,58 @@ func runServer(server *http.Server, cancel context.CancelFunc) {
 	defer cancel()
 	defer shutdownSentry() // shutdownSentry must be executed in the main goroutine.
 
-	log.WithField("address", server.Addr).Info("Starting server")
+	httpListeners := getHTTPListeners(server)
+	serveHTTPListeners(server, httpListeners)
+}
+
+func getHTTPListeners(server *http.Server) (httpListeners []net.Listener) {
+	var err error
+	if config.Config.Server.SystemdSocketActivation {
+		httpListeners, err = activation.Listeners()
+	} else {
+		var httpListener net.Listener
+		httpListener, err = net.Listen("tcp", server.Addr)
+		httpListeners = append(httpListeners, httpListener)
+	}
+	if err != nil || httpListeners == nil || len(httpListeners) == 0 {
+		log.WithError(err).
+			WithField("listeners", httpListeners).
+			WithField("systemd_socket", config.Config.Server.SystemdSocketActivation).
+			Fatal("Failed listening to any socket")
+		return nil
+	}
+	return httpListeners
+}
+
+func serveHTTPListeners(server *http.Server, httpListeners []net.Listener) {
+	var wg sync.WaitGroup
+	wg.Add(len(httpListeners))
+	for _, l := range httpListeners {
+		go func(listener net.Listener) {
+			defer wg.Done()
+			log.WithField("address", listener.Addr()).Info("Serving Listener")
+			serveHTTPListener(server, listener)
+		}(l)
+	}
+	wg.Wait()
+}
+
+func serveHTTPListener(server *http.Server, l net.Listener) {
 	var err error
 	if config.Config.Server.TLS.Active {
 		server.TLSConfig = config.TLSConfig
-		log.
-			WithField("CertFile", config.Config.Server.TLS.CertFile).
+		log.WithField("CertFile", config.Config.Server.TLS.CertFile).
 			WithField("KeyFile", config.Config.Server.TLS.KeyFile).
 			Debug("Using TLS")
-		err = server.ListenAndServeTLS(config.Config.Server.TLS.CertFile, config.Config.Server.TLS.KeyFile)
+		err = server.ServeTLS(l, config.Config.Server.TLS.CertFile, config.Config.Server.TLS.KeyFile)
 	} else {
-		err = server.ListenAndServe()
+		err = server.Serve(l)
 	}
-	if err != nil {
-		if errors.Is(err, http.ErrServerClosed) {
-			log.WithError(err).Info("Server closed")
-		} else {
-			log.WithError(err).Fatal("Error during listening and serving")
-		}
+
+	if errors.Is(err, http.ErrServerClosed) {
+		log.WithError(err).WithField("listener", l).Info("Server closed")
+	} else {
+		log.WithError(err).WithField("listener", l).Error("Error during listening and serving")
```

Comment:
Here, the log output contains a reference to the listener, rather than the plain text details. Is that desired:

```
time="2024-01-12T22:05:03.997280Z" level=info msg="Loading environments" package=environment
time="2024-01-12T22:05:04.027566Z" level=info msg="Loading runners" package=runner
time="2024-01-12T22:05:04.027605Z" level=info msg="First Recovery Done" package=main
time="2024-01-12T22:05:04.027609Z" level=info msg="Watching Event Stream" package=runner
time="2024-01-12T22:05:04.027976Z" level=info msg="Serving Listener" address="127.0.0.1:7200" package=main

^Ctime="2024-01-12T22:05:23.226584Z" level=info msg="Received SIGINT, shutting down..." package=main
time="2024-01-12T22:05:23.226787Z" level=info msg="Server closed" error="http: Server closed" listener="&{0x14000508100 {<nil> 0 0}}" package=main

Process finished with the exit code 0
```